pkg/volume/csi/csi_attacher.go

```diff
@@ -595,14 +595,13 @@ func (c *csiAttacher) UnmountDevice(deviceMountPath string) error {
 		driverName = data[volDataKey.driverName]
 		volID = data[volDataKey.volHandle]
 	} else {
-		klog.Error(log("UnmountDevice failed to load volume data file [%s]: %v", dataDir, err))
-
-		// The volume might have been mounted by old CSI volume plugin. Fall back to the old behavior: read PV from API server
```

Comment:
I think `vol_data.json` was brought in https://github.com/kubernetes/kubernetes/pull/64882 5 years ago, no need to fall back to the old behavior when it does not exist, it should be safe to remove.

Reply:
shall we move this PR forward since it really solves the problem per our large scale test, there is no disk detach issue any more. @mattcary @saikat-royc @msau42

Reply:
https://github.com/kubernetes/kubernetes/pull/107065 is the PR where the new global path is introduced (1.24). The PR states that the node need to be drained explicitly in its release note. ```Change node staging path for csi driver to use a PV agnostic path. Nodes must be drained before updating the kubelet with this change.```

So I think if vol_data.json does not exist, we can skip the else part of the code to parse the path. [Note: this also means that this PR can be cherry-picked to >= 1.24, since we already have a dependency to drain node when upgrading from < 1.24 to 1.24]. So +1 from my side to remove the fallback mountpath parsing